src/Factories/Transform/UserSchema/OptionalIfTransformFactory.cs

```diff
@@ -0,0 +1,38 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using form_builder.Conditions;
+using form_builder.Models;
+using form_builder.Models.Elements;
+
+namespace form_builder.Factories.Transform.UserSchema
+{
+    public class OptionalIfTransformFactory : IUserPageTransformFactory
+    {
+        public async Task<Page> Transform(Page page, FormAnswers convertedAnswers)
+        { 
+            IEnumerable<IElement> elements = page.ValidatableElements
+                .Where(element => !string.IsNullOrEmpty(element.Properties.OptionalIf?.QuestionId));
```

Comment:
Do you still need the null check now? The "?" on OptionalIf... If we've switched to validatable elements, won't they always have Properties, and therefore always get a new OptionalIf, which would have a null QuestionId, but that's okay. I know it was having issues on the "Properties" of some elements.